packages/cozy-client/src/models/file.js

```diff
@@ -291,3 +292,268 @@ export const isReferencedByAlbum = file => {
+ * @param   {boolean} force                - Whether we should overwrite the destination in case of conflict (defaults to false)
+ * @returns {Promise}                     - A promise that returns the move action response and the deleted file id (if any) if resolved or an Error if rejected
+ *
+ */
+export const move = async (client, fileId, destination, force = false) => {
+  const { folderId, path } = destination
+  try {
+    const resp = await client
+      .collection(DOCTYPE_FILES)
+      .updateFileMetadata(fileId, {
+        dir_id: folderId
+      })
+
+    return {
+      moved: resp.data,
+      deleted: null
+    }
+  } catch (e) {
+    if (e.status === 409 && force) {
+      let destinationPath
+      if (path) {
+        destinationPath = path
+      } else {
+        const { data: movedFile } = await client.query(
+          Q(DOCTYPE_FILES).getById(fileId)
+        )
+        const filename = movedFile.name
+        destinationPath = await getFullpath(client, folderId, filename)
+      }
+      const conflictResp = await client
+        .collection(DOCTYPE_FILES)
+        .statByPath(destinationPath)
+      await client.collection(DOCTYPE_FILES).destroy(conflictResp.data)
+      const resp = await client
+        .collection(DOCTYPE_FILES)
+        .updateFileMetadata(fileId, {
+          dir_id: folderId
+        })
+
+      return {
+        moved: resp.data,
+        deleted: conflictResp.data.id
+      }
+    } else {
+      throw e
+    }
+  }
+}
+
+/**
+ *
+ * Method to upload a file even if a file with the same name already exists.
+ *
+ * @param {CozyClient} client - The CozyClient instance
+ * @param {string} pathArg Fullpath for the file ex: path/to/
+ * @param {object} file HTML Object file
+ * @param {Object} metadata An object containing the wanted metadata to attach
+ */
+export const overrideFileForPath = async (client, pathArg, file, metadata) => {
+  let path = pathArg
+  if (!path.endsWith('/')) path = path + '/'
+
+  const filesCollection = client.collection(DOCTYPE_FILES)
+  try {
+    const existingFile = await filesCollection.statByPath(path + file.name)
+
+    const { id: fileId, dir_id: dirId } = existingFile.data
+    const resp = await filesCollection.updateFile(file, {
+      dirId,
+      fileId,
+      metadata
+    })
+    return resp
+  } catch (error) {
+    if (/Not Found/.test(error)) {
+      const dirId = await filesCollection.ensureDirectoryExists(path)
+      const createdFile = await filesCollection.createFile(file, {
+        dirId,
+        metadata
+      })
+      return createdFile
+    }
+    throw error
+  }
+}
+/**
+ * Method to generate a new filename if there is a conflict
+ *
+ * @param {string} filenameWithoutExtension A filename without the extension
+ * @returns {string} A filename with the right suffix
+ */
+export const generateNewFileNameOnConflict = filenameWithoutExtension => {
+  //Check if the string ends by _1
+  const regex = new RegExp('(_)([0-9]+)$')
```

Comment:
You may want a look-behind assertion for matching the `_`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions#other_assertions